src/vcml/models/generic/fbdev.cpp

```diff
@@ -20,6 +20,162 @@ void fbdev::update() {
     }
 }
 
+template <typename T>
+void write_to_ostream(ostream& os, const T* val) {
+    os.write(reinterpret_cast<const char*>(val), sizeof(T));
+}
+
+struct bmp_header {
```

Comment:
I would prefer to have the headers as PODs. So no methods or constructors; you can replace header_size with sizeof(header); and initialize the header field when you need them, e.g.
```
bmp_header header{};
header.size = sizeof(header);
header.head[0] = 'B';
header.head[1] = 'M';
...
```
Maybe add a `static_assert` for the size of the structs to make sure no padding has happened...